spec.html

```diff
@@ -39308,54 +39224,24 @@ <h1>Promise.prototype.finally ( _onFinally_ )</h1>
             1. Let _thenFinally_ be _onFinally_.
             1. Let _catchFinally_ be _onFinally_.
           1. Else,
-            1. Let _stepsThenFinally_ be the algorithm steps defined in <emu-xref href="#sec-thenfinallyfunctions" title></emu-xref>.
-            1. Let _lengthThenFinally_ be the number of non-optional parameters of the function definition in <emu-xref href="#sec-thenfinallyfunctions" title></emu-xref>.
-            1. Let _thenFinally_ be ! CreateBuiltinFunction(_stepsThenFinally_, _lengthThenFinally_, *""*, &laquo; [[Constructor]], [[OnFinally]] &raquo;).
-            1. Set _thenFinally_.[[Constructor]] to _C_.
-            1. Set _thenFinally_.[[OnFinally]] to _onFinally_.
-            1. Let _stepsCatchFinally_ be the algorithm steps defined in <emu-xref href="#sec-catchfinallyfunctions" title></emu-xref>.
-            1. Let _lengthCatchFinally_ be the number of non-optional parameters of the function definition in <emu-xref href="#sec-catchfinallyfunctions" title></emu-xref>.
-            1. Let _catchFinally_ be ! CreateBuiltinFunction(_stepsCatchFinally_, _lengthCatchFinally_, *""*, &laquo; [[Constructor]], [[OnFinally]] &raquo;).
-            1. Set _catchFinally_.[[Constructor]] to _C_.
-            1. Set _catchFinally_.[[OnFinally]] to _onFinally_.
+            1. Let _thenFinallyClosure_ be a new Abstract Closure with parameters (_value_) that captures _onFinally_ and _C_ and performs the following steps when called:
+              1. Let _result_ be ? Call(_onFinally_, *undefined*).
+              1. Let _promise_ be ? PromiseResolve(_C_, _result_).
+              1. Let _returnValue_ be a new Abstract Closure with no parameters that captures _value_ and performs the following steps when called:
+                1. Return _value_.
+              1. Let _valueThunk_ be ! CreateBuiltinFunction(_returnValue_, 0, *""*, &laquo; &raquo;).
+              1. Return ? Invoke(_promise_, *"then"*, &laquo; _valueThunk_ &raquo;).
+            1. Let _thenFinally_ be ! CreateBuiltinFunction(_thenFinallyClosure_, 1, *""*, &laquo; &raquo;).
+            1. Let _catchFinallyClosure_ be a new Abstract Closure with parameters (_reason_) that captures _onFinally_ and _C_ and performs the following steps when called:
+              1. Let _result_ be ? Call(_onFinally_, *undefined*).
+              1. Let _promise_ be ? PromiseResolve(_C_, _result_).
+              1. Let _throwReason_ be a new Abstract Closure with no parameters that captures _reason_ and performs the following steps when called:
+                1. Return ThrowCompletion(_reason_).
+              1. Let _thrower_ be ! CreateBuiltinFunction(_throwReason_, 0, *""*, &laquo; &raquo;).
+              1. Return ? Invoke(_promise_, *"then"*, &laquo; _thrower_ &raquo;).
+            1. Let _catchFinally_ be ! CreateBuiltinFunction(_catchFinallyClosure_, 1, *""*, &laquo; &raquo;).
```

Comment:
fwiw, confirmed these exact changes pass in https://npmjs.com/promise.prototype.finally